Connection: protected owner errors manual fix

bindlegirl opened the pull request:
Related to: [VULCAN-86](https://linear.app/a8c/issue/VULCAN-86)
P2: pghT6q-3V-p2

<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Adding filter to Error_Handler class so wpcomsh can plug in with Protected owner errors.
* Adapted js components to handle protected owner errors and display a different button that redirects users to account creation.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Testing instructions in https://github.com/Automattic/jetpack/pull/43601

bindlegirl committed:
Added protected owner error support

bindlegirl committed:
Changelogs

bindlegirl committed:
Fix for my jetpack

bindlegirl committed:
changelog

bindlegirl committed:
Add tests

bindlegirl committed:
Everything goes to error_handler

bindlegirl committed:
Bug hunting and tests

bindlegirl committed:
Revert change in initial state file

bindlegirl committed:
Remove redundant js component

fgiannar commented on projects/js-packages/connection/hooks/use-connection-error-notice/index.jsx:
```diff
@@ -11,7 +11,9 @@ import useRestoreConnection from '../../hooks/use-restore-connection/index.jsx';
  */
 export default function useConnectionErrorNotice() {
 	const { connectionErrors } = useConnection( {} );
+
```
Do we need these changes here?

fgiannar commented on projects/packages/connection/src/class-error-handler.php:
```diff
@@ -156,29 +156,40 @@ private function __construct() {
+			'invalid_connection_owner',
+		);
+
+		// Automatically include any error codes that aren't in the default list
```
As we discussed, I think it's safer to hardcode the error code coming from `wpcomsh` here and also add it in the `$known_errors` list.
The concept of this class is that we only allow verified connection errors to show up.
An exception is the `invalid_connection_owner` which is not verified but is set by the same package so it's safe.
The filter we introduced is enough to allow `wpcomsh` to handle it's own connection errors without storing them but I don't see any other plugin handling connection errors.

bindlegirl replied:
Changed in latest comit

bindlegirl committed:
Hardcoding protected owner error to verified error array

bindlegirl committed:
Unset invalid_connection_owner on WoA

bindlegirl committed:
Remove WoA handling so we can move it to wpcomsh

fgiannar reviewed (changes requested):
This is what I see when I create the missing user manually:

<img width="1131" alt="Screenshot 2025-05-29 at 15 28 58" src="https://github.com/user-attachments/assets/371b7d6a-a09f-46c3-b6b9-e28b4907438d" />

I was testing with a secondary **non-connected** admin. I was under the impression that we only display errors for the current user when the error has `user_id` set. Isn't this the case?

bindlegirl committed:
Updated my jetpack and react dshboard error messages

bindlegirl committed:
Update tests

bindlegirl commented:
I have cleared all the errors in the other PR and updated error messages for the React dashboard so that only the protected owner gets a custom message. I have also updated the My Jetpack code to follow the updated logic and not depend on error messages.

bindlegirl committed:
Update messaging method and tests

bindlegirl committed:
Remove other errors from cxn component when owner error exists

bindlegirl committed:
Fix test

fgiannar reviewed (dismissed):
Amazing work, Ana!

Thank you for working on this and applying all the suggested recommendations 👍 

I followed the test instructions and can confirm that I didn't see any Connection errors apart from the missing site owner error, which immediately went away after creating the corresponding local user.

bindlegirl committed:
Edit notice component

bindlegirl committed:
Update js tests

bindlegirl committed:
More js tests

bindlegirl commented:
Hey @fgiannar, I made some changes to the JS files so they make more sense 🤞 
Could you take this PR for one more ride?

bindlegirl committed:
Adding support for pre-populating emails
